test/test_csvs.py

```diff
@@ -0,0 +1,253 @@
+            if index in fields_to_validate:
+                # the common fields validated by a collection of values
+                #  (ReferenceDesignator, _ParameterID_R/T, _Units)
+                if header[index] in validation_dictionary:
+                    validation_collection = validation_dictionary[header[index]]
+                    if column not in validation_collection:
+                        invalid_entries.append(header[index] + ': [' + column + ']')
+                # the common fields validated by a portion of the ReferenceDesignator field
+                elif header[index] in validation_positions:
+                    begin,end = validation_positions[header[index]]
+                    column_index = header.index('ReferenceDesignator')
+                    if column != line[column_index][begin:end]:
+                        invalid_entries.append(header[index] + ': [' + column + ']')
+                # the global and gradient fields
+                elif header[index] in ('GlobalRangeMin','GlobalRangeMax','GradientTest_mindx'):
+                    if not self._is_empty_or_numeric(column):
+                        invalid_entries.append(header[index] + ': [' + str(column) + ']')
+                # the spike fields
+                elif header[index] in ('SpikeTest_ACC','SpikeTest_N','SpikeTest_L'):
+                    if not self._is_empty_or_numeric(column):
+                        invalid_entries.append(header[index] + ': [' + str(column) + ']')
+                # the stuck fields
+                elif header[index] in ('StuckValueTest_ResolutionR','StuckValueTest_NumRepeatValues'):
+                    if not self._is_empty_or_numeric(column):
+                        invalid_entries.append(header[index] + ': [' + str(column) + ']')
+                # the trend fields
+                elif header[index] in ('TrendTest_TimeIntLengthDays','TrendTest_PolynomialOrder','TrendTest_nstd'):
+                    if not self._is_empty_or_numeric(column):
+                        invalid_entries.append(header[index] + ': [' + str(column) + ']')
+
+        return invalid_entries
+
+    def test_validate_global_range_file(self):
+        valid_file = True # file is valid until proven otherwise
+        # this gets the file's header line
+        f = pd.read_csv(self._get_path('../data_qc_global_range_values.csv'),
+                        encoding=self._encoding, na_filter=False)
+        # this gets the file's header line
+        header = list(f)
+        # all the numbers of the fields in each data record needing validation
+        fields_to_validate = range(len(header)) # validate all fields
+
+        for index,line in f.iterrows():
+            invalid_list = self._validate_line(header, fields_to_validate, line)
+            # if it contains entries the file is invalid
+            if invalid_list:
+                valid_file = False
+                print('Global record invalid ' + str(index+1) + ', ')
```

Comment:
I'm not sure if the print statements will play nicely with the nosetest. Have you looked at the output?

Reply:
Yes. It looks fine.